client/my-sites/promote-post-i2/components/campaign-item-details/index.tsx

```diff
@@ -186,6 +308,67 @@ export default function CampaignItemDetails( props: Props ) {
 	const deliveryEstimateFormatted = getCampaignEstimatedImpressions( display_delivery_estimate );
 	const campaignTitleFormatted = title || __( 'Untitled' );
 	const campaignCreatedFormatted = moment.utc( created_at ).format( 'MMMM DD, YYYY' );
+	const objectiveFormatted = ( () => {
```

Comment:
We are working on handling the objectives in the DSP Sever so that we can modify them without updating all of our clients. 

It would be nice to get this information for the DSP server on the campaign details endpoint for this specific case.  
The code resides in the `v1.1/campaign/objectives` endpoint (DSP code).

What do you think? We can go with this alternative, but we will need to update it in the future. 

Feel free to ignore this comment if you wan this shipped right now.

Reply:
Please check again @sbarbosa 

i added the campaign/objectives endpoint

I had to alter a bit the function because it is the 1st endpoint v1.1 we load into calypso

I am also not sure about the locale.. calling the endpoint with different locale values doesn't seem to work (it wont bring translated values)

Reply:
Thanks for the change, @therocket-gr.
For this functionality, returning the data in the campaign details endpoint would be more straightforward.

For example, we could include another prop, `objectiveDescription` with the information about the objective only for the campaign details endpoint.

I am thinking that in the future we can add/delete some of the objectives from the list. If we do that, with the current approach, we won't be able to get the objective data (it won't be present in the `/campaign/objectives endpoint`.

Reply:
i added a new object in the DSP side when querying the campaign data
```
"objective_data": {
    "title": string,
    "description": string,
    "suitable_for_description": string
  },
```

in the client I am fetching the object and treating it (so that if undefined it wont render the objective block and to get the right icon to display if available

you can check the undefined (not displaying the objective) when checking an old campaign that either doesn't have anything in the objective_data or the objective is "views" which is an old flag and wont bring any details

@j6ll  and @sbarbosa can you please retest this PR and do a CR as the code has changed significantly